### DISCUSSION TITLE
It seems getAvailablePurchases() , availablePurchases is not returning currently subscribed products on iOS 15 or higher.

### DISCUSSION THREAD

**Discussion by dmk3141618** [hyochan/react-native-iap#2626] [Q&A] [answered]
Is there a way to determine what subscriptions are currently active on iOS 15 or higher? I am currently working with STOREKIT2 and versions above iOS 15. When I call getAvailablePurchases() and check availablePurchases, it doesn't show the subscription products I am currently subscribed to, even though I have active subscriptions. It seems that the currently subscribed products appear correctly on Android.

When I call getPurchaseHistory() and check purchaseHistory, it seems to show all the purchase histories based on the transactionId of the subscription products. Should I use this information to take the latest transaction history of the subscription products with the same originalTransactionId and then query related information through the App Store Server API?

In summary, the question is whether it's possible to know what is currently being subscribed to. I need this information to implement a feature for restoring purchase of users. Currently, I have implemented it by calling requestSubscription for all subscription products using asyncForEach since I do not know what is currently subscribed. Without knowing the current subscriptions, the only method I had was to try requestSubscription on all the subscription products.

**Comment by dmk3141618** [comment]
Is there any method which can use Transaction.currentEntitlements or Product.SubscriptionInfo of STOREKIT2 ?

**Comment by dmk3141618** [comment, marked as answer]
I've found these methods. It works well. Thank you.

IapIosSk2.subscriptionStatus(itemSubs[0])
IapIosSk2.currentEntitlement(itemSubs[0])
IapIosSk2.latestTransaction(itemSubs[0])
